frontend/helpers/product-list-helpers.ts

```diff
@@ -118,18 +118,19 @@ export function getAdminLinks({
    ];
 }
 
-export function updateProductListWithItemTypeOverrides(
+export function calculateProductListOverrides(
    productList: ProductList,
    page: number,
    itemType?: string
 ): ProductList {
-   const productListCopy = { ...productList };
-   productListCopy.title = getTitle(productList, itemType);
-   productListCopy.metaTitle = getMetaTitle(productList, itemType);
-   productListCopy.description = getDescription(productList, page, itemType);
-   productListCopy.metaDescription = getMetaDescription(productList, itemType);
-   productListCopy.tagline = getTagline(productList, page, itemType);
-   return productListCopy;
+   productList.overrides = {
```

Comment:
I'm not sure I understand why we're back to modifying the original. I think this will introduce rendering problems by modifying the props. If we are intentionally modifying it, this should be a void function, but I think this is asking for trouble.